test/liquid/combinators/tags/if_test.exs

```diff
@@ -0,0 +1,170 @@
+defmodule Liquid.Combinators.Tags.IfTest do
+  use ExUnit.Case
+
+  import Liquid.Helpers
+  alias Liquid.NimbleParser, as: Parser
+
+  test "if using booleans" do
+    test_combinator(
+      "{% if false %} this text should not go into the output {% endif %}",
+      &Parser.if/1,
+      [{:if, ["false", " this text should not go into the output "]}]
+    )
+
+    test_combinator(
+      "{% if true %} this text should go into the output {% endif %}",
+      &Parser.if/1,
+      [{:if, ["true", " this text should go into the output "]}]
+    )
+  end
+
+  test "if else " do
+    test_combinator("{% if \"foo\" %} YES {% else %} NO {% endif %}", &Parser.if/1, [
+      {:if, ["foo", " YES ", {:else, [" NO "]}]}
+    ])
+  end
+
+  test "opening if tag with multiple conditions " do
+    test_combinator(
+      "{% if line_item.grams > 20000 and customer_address.city == 'Ottawa' or customer_address.city == 'Seatle' %}hello test{% endif %}",
+      &Parser.if/1,
+      if: [
```

Comment:
I don't think the parsing of this produces a correct AST. In order to evaluate it properly it would need another pass. 

I think its better to make a nested AST in order of precedence:
```
{
  :or,
  {
    :and,
    {
      :>, # (Or :greater_than, which is defined as token for operators)
      {:variable, ["line_item", "grams"]},
      20000
    },
    {
      :==, # (Or :equals)
      {:variable, ["customer_address", "city"]},
      'Ottawa'
    }
  },
  {
    :==, 
    {:variable, ["customer_address", "city"]},
    'Seatle'
  }
}
```

`:or` has precedence on `:and` in both elixir and Ruby, So `:or` has to be parsed first:
```
iex(1)> true and false or true
true
```

Also note that expected tokens should be presented in the AST as Atoms, not binaries, to be more memory efficient.

Reply:
I think that `and` is higher precedence than `or` (`ands` always is evaluated before `ors`)  and this is the reason for `or` be _outside_ `and`. Great point, we didn't think about precedence order. The `atom` think is important too

Reply:
Actually in both Elixir's and Ruby's logical operators `or` has precedence over `and`:

```
iex(2)> quote do
...(2)> true and false or true
...(2)> end
{:or, [context: Elixir, import: Kernel],
 [{:and, [context: Elixir, import: Kernel], [true, false]}, true]}
```